templates/deployment.yaml

```diff
@@ -14,18 +14,17 @@ spec:
       app: {{ template "docker-registry.name" . }}
       release: {{ .Release.Name }}
   replicas: {{ .Values.replicaCount }}
-{{- if .Values.updateStrategy }}
-  strategy:
-{{ toYaml .Values.updateStrategy | indent 4 }}
-{{- end }}
+  {{- if .Values.updateStrategy }}
+  strategy: {{ toYaml .Values.updateStrategy | nindent 4 }}
+  {{- end }}
   minReadySeconds: 5
   template:
     metadata:
       labels:
         app: {{ template "docker-registry.name" . }}
         release: {{ .Release.Name }}
-        {{- if .Values.podLabels }}
-{{ toYaml .Values.podLabels | indent 8 }}
+        {{- with .Values.podLabels }}
+        {{ toYaml . | nindent 8 }}
```

Comment:
Can you elaborate on the difference between the before/after here? I'm mainly just curious, since the difference, if any, seems either stylistic or quite nuanced, and I'd like to understand:

```yaml
{{- if .Values.podLabels }}
{{ toYaml .Values.podLabels | indent 8 }}
{{- end }}
```

...vs...

```yaml
{{- with .Values.podLabels }}
{{ toYaml . | nindent 8 }}
{{- end }}
```

Is one form or the other more idiomatic? Is there a subtle whitespace difference in the two outputs? This could affect the way I personally write conditionals like this, hence the endeavor to learn.

Reply:
Maybe @kuzaxak can add to the reasoning behind the change, but here are my two cents:

As from [docs](https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with) I think `with` will mainly result in shorter statements when accessing multiple objects in a nested structure, and arguably increases readability.

I find the `if` variation in this simple form just as adequate and I don't think performance (multiple lookups) is of any relevance here. Often you won't need the `else` option and the both are interchangeable, where `if` probably feels more familiar for devs coming from functional languages.

Reply:
Thanks for the $0.02! Both versions seem fine. In my view, there doesn't seem to be an adequately convincing argument to use one over the other, but it's also trivial enough and part of a larger changeset that the choice doesn't carry much weight either way. :+1:

Reply:
Hi,

As you said with works in the same way as in Python. If you have nested object somewhere and you want to change pointer for the current context `.` you can use `with`.

But keep that in mind if you need to access anything outside of the context you need to use root context via `$.`.

Main goal is reduce duplicating nested path, as an example:

To render something like `.worker.ingresses.first.annotations` it would make sense to use `with`